### PR TITLE
Plot coverage

### PR DESCRIPTION
**Pull request by bgruening**
Any review appreciated ...

ToDo:
* Help needs to be updated
* checking for usability
* implementing `--plotFileFormat`

ping @steffenheyne

### CODE REVIEW

File: galaxy/wrapper/plotCoverage.xml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                --numberOfSamples $advancedOpt.numberOfSamples
+
+                #if $advancedOpt.plotTitle and str($advancedOpt.plotTitle.value) != "":
+                    --plotTitle '$advancedOpt.plotTitle'
+                #end if
+                @ADVANCED_OPTS_READ_PROCESSING@
+            #end if
+        &&
+
+        mv plot.png '$plotFile'
```
Comment (bgruening) [inline galaxy/wrapper/plotCoverage.xml]: This is why a `--plotFileFormat` option would be useful.

File: galaxy/wrapper/plotCoverage.xml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                <expand macro="skipZeros" />
+                <expand macro="plotTitle" />
+            </when>
+        </conditional>
+        <param argument="--outRawCounts" type="boolean" label="Save raw counts (coverages) to file" help=""/>
+
+
+    </inputs>
+    <outputs>
+        <data format="png" name="plotFile" label="${tool.name} on ${on_string}: bin counts" />
```
Comment (dpryan79) [inline galaxy/wrapper/plotCoverage.xml]: I imagine that this would be where `--plotFileFormat` would be most useful. Do conditionals work on the output?

Comment (bgruening) [inline galaxy/wrapper/plotCoverage.xml]: Yes they do: https://github.com/fidelram/deepTools/blob/master/galaxy/wrapper/deepTools_macros.xml#L333
I will be afk for 4h ... train ...

Comment (dpryan79) [inline galaxy/wrapper/plotCoverage.xml]: Let me know if this is a WIP or not later today. Enjoy the train :)

**Comment by bgruening** [comment]
This sould fix https://github.com/fidelram/deepTools/issues/185 and https://github.com/fidelram/deepTools/issues/180. And it includes a test for plotCoverage.

Let me know if anything is missing and feel free to merge @dpryan79. Internet is soooo sloooow.

**Comment by bgruening** [comment]
Tests passing for me locally ... anyone with this conda install from travis mind to regenerate the test data?
I do have matplotlib 1.5.0, so no clue what happens here.

**Comment by dpryan79** [comment]
I'll have a look. Every version of matplotlib likes to be its own special snowflake when it comes to formatting SVGs.

**Comment by dpryan79** [comment]
I've tried 1.5.0 and 1.4.1, I can get things to work locally, but not on travis. I'll revert my last commit.

**Comment by dpryan79** [comment]
I have absolutely no clue why that worked, it has 5 failures locally!

**Comment by bgruening** [comment]
Same for me ... this is strange!
Can we make the svg comparison function more tolerant?

**Comment by bgruening** [comment]
But hey its green at it fixes two issues :) Want to merge?

**Comment by dpryan79** [comment]
I hope so, I haven't really looked to see how it works. I'll merge and have @steffenheyne have a look after I update our local instance tomorrow.

**Pull request merged**